Source/CarthageKit/GitHub.swift

```diff
@@ -266,6 +284,14 @@ internal struct GitHubCredentials {
 	}
 }
 
+internal func loadGitHubCredentials(accessToken: GitHubAccessToken?) -> SignalProducer<GitHubCredentials?, CarthageError> {
```

Comment:
@mdiep I tried to implement the `GitHubCredentials` enum but after doing the refactor I got a compiler seg fault. The problem is with `internal func loadGitHubCredentials() -> SignalProducer<GitHubCredentials?, CarthageError>`. When I try to invoke this method in https://github.com/guidomb/Carthage/blob/github-access-token-option/Source/CarthageKit/Project.swift#L310 I get a seg fault from the compiler. Apparently the compiler cannot handle a generic type paremeter of an optional enum.

So we have two options: I either change the method signature to `internal func loadGitHubCredentials() -> SignalProducer<Box<GitHubCredentials>?, CarthageError>` or I do what you originally suggested (maybe you can see the future :-P) and delete anything related to `GitHubCredentials`, change the method signature to `internal func loadGitHubAuthorization() -> SignalProducer<String?, CarthageError>` and change the method that received a `GitHubCredential` to receive the value of the authorization header (which I still think is not a good option)

Another option is to change `downloadAsset` and `releaseForTag` not to receive the credentials and load them inside those method. The only problem is that when somebody uses this two methods the credentials are going to be loded more than once. This actually happens inside `installBinariesForProject` in `Project.swift`.

What do you think is best?